ipld/hamt/src/node.rs

```diff
@@ -0,0 +1,388 @@
+                    let kvs = std::mem::replace(vals, Vec::new());
+                    for p in kvs.into_iter() {
+                        let hash = Self::hash(p.key());
+                        sub.modify_value(
+                            &mut HashBits::new_at_index(&hash, consumed),
+                            bit_width,
+                            depth + 1,
+                            p.0,
+                            p.1,
+                            store,
+                        )?;
+                    }
+
+                    *child = Pointer::Cache(Box::new(sub));
+                    return Ok(None);
+                }
+
+                // Otherwise insert the element into the array in order.
+                let max = vals.len();
+                let idx = vals
+                    .iter()
+                    .position(|c| c.key() > &key)
+                    .unwrap_or_else(|| max);
+
+                let np = KeyValuePair::new(key, value);
+                vals.insert(idx, np);
+
+                Ok(None)
+            }
+        }
+    }
+
+    /// Internal method to delete entries.
+    pub fn rm_value<Q: ?Sized, S: BlockStore>(
+        &mut self,
+        hashed_key: &mut HashBits,
+        bit_width: u8,
+        depth: usize,
+        key: &Q,
+        store: &S,
+    ) -> Result<Option<(K, V)>, Error>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        let idx = hashed_key.next(bit_width)?;
+
+        // No existing values at this point.
+        if !self.bitfield.test_bit(idx) {
+            return Ok(None);
+        }
+
+        let cindex = self.index_for_bit_pos(idx);
+        let child = self.get_child_mut(cindex);
+
+        match child {
+            Pointer::Link(cid) => match store.get::<Node<K, V>>(cid)? {
+                Some(mut node) => {
+                    // Pull value from store and update to cached node
+                    let del = node.rm_value(hashed_key, bit_width, depth + 1, key, store)?;
+                    *child = Pointer::Cache(Box::new(node));
+
+                    // Clean to retrieve canonical form
+                    child.clean()?;
+                    Ok(del)
+                }
+                None => Err(Error::Custom("node not found")),
+            },
+            Pointer::Cache(n) => {
+                // Delete value and return deleted value
+                let deleted = n.rm_value(hashed_key, bit_width, depth + 1, key, store)?;
+
+                // Clean to ensure canonical form
+                child.clean()?;
+                Ok(deleted)
+            }
+            Pointer::Values(vals) => {
+                // Delete value
+                for (i, p) in vals.iter().enumerate() {
+                    if key.eq(p.key().borrow()) {
+                        let old = if vals.len() == 1 {
+                            if let Pointer::Values(new_v) = self.rm_child(cindex, idx) {
+                                new_v.into_iter().nth(0).unwrap()
```

Comment:
Not sure why it's not showing up in our CI check, however I am getting a Clippy error on this line: 

```
error: called `.nth(0)` on a `std::iter::Iterator`
   --> ipld/hamt/src/node.rs:330:33
    |
330 | ...                   new_v.into_iter().nth(0).unwrap()
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^ help: try calling: `new_v.into_iter().next()`
    |
    = note: `-D clippy::iter-nth-zero` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#iter_nth_zero
```

Reply:
from updated rust toolchain I'm seeing it now, nice catch